Helper/RateLimiter.php

```diff
@@ -52,38 +52,49 @@ class RateLimiter
      */
     const POWER = 2;
 
+    /**
+     * Number of allowed notification requests
+     */
+    const NUMBER_OF_ATTEMPTS = 6;
+
 
     /**
      * RateLimiter constructor.
      *
      * @param CacheInterface $cache
      * @param SerializerInterface $serializer
      * @param RemoteAddress $remoteAddress
-     * @param SessionManagerInterface $sessionManager
+     * @param AdyenLogger $adyenLogger
      */
 
     public function __construct(
         CacheInterface $cache,
         SerializerInterface $serializer,
         RemoteAddress $remoteAddress,
-        SessionManagerInterface $sessionManager
+        AdyenLogger $adyenLogger
     ) {
         $this->cache = $cache;
         $this->serializer = $serializer;
         $this->remoteAddress = $remoteAddress;
-        $this->sessionManager = $sessionManager;
+        $this->adyenLogger = $adyenLogger;
     }
 
 
     private function getCacheId()
     {
-        return "adyen-logins-" . $this->sessionManager->getSessionId() . "-" . $this->remoteAddress->getRemoteAddress();
+        return "adyen-logins-" . $this->remoteAddress->getRemoteAddress();
     }
 
     public function saveSessionIdIpAddressToCache()
     {
         $cacheValue = $this->getNumberOfAttempts();
 
+        if($cacheValue === self::NUMBER_OF_ATTEMPTS) {
+            $this->adyenLogger->addAdyenDebug(
+                "Notification has been rejected because the allowed number of attempts is exceeded."
```

Comment:
Can we adjust the message and add the IP address by using sprintf()?
```suggestion
                "Webhook has been rejected because the allowed number of authentication attempts has been exceeded."
```

Reply:
🙏